cilium/cmd/identity_list.go

```diff
@@ -38,28 +38,51 @@ func init() {
 	identityCmd.AddCommand(identityListCmd)
 	identityListCmd.Flags().BoolVarP(&reservedIDs, "reserved", "", false,
 		"List all reserved identities")
+	AddJSONSupport(identityListCmd)
 }
 
 func listIdentities(args []string) {
+	result := make(map[string]interface{})
+
 	if reservedIDs {
-		fmt.Println("Reserved identities:")
+		reservedMap := make(map[string]string)
 		for k, v := range policy.ReservedIdentities {
-			fmt.Printf("%3d %-15s \n", v, k)
+			reservedMap[fmt.Sprintf("%d", v)] = k
 		}
+		result["reservedIDs"] = reservedMap
 	}
-	fmt.Println("Identities in use by endpoints:\n" +
-		"(Note: If labels have been provided as parameters, only matching identities will be displayed)")
 
 	var params *identityApi.GetIdentityParams
 	if len(args) != 0 {
 		params = identityApi.NewGetIdentityParams().WithLabels(args)
 	}
 
-	if identities, err := client.Policy.GetIdentity(params); err != nil {
+	identities, err := client.Policy.GetIdentity(params)
+	if err != nil {
 		Fatalf("Cannot get identities for given labels %v. err: %s\n", params.Labels, err.Error())
-	} else if b, err := json.MarshalIndent(identities, "", "  "); err != nil {
+	}
+	result["identities"] = identities.Payload
+
+	if JSONOutput || len(JSONPath) > 0 {
+		if err := DumpJSON(result, JSONPath); err != nil {
+			os.Exit(1)
```

Comment:
We can just `os.Exit(1)` from `DumpJSON` so we don't need to duplicate the error handling everywhere

Reply:
Ok, it was like that, but @nebril requested to change the function to return error instead exit. 

I'll change again.

Reply:
I've read through @nebril's comment and I agree with the testing aspect. I'm OK to keep it as-is.